Zend/zend_closures.c

```diff
@@ -377,10 +377,49 @@ static ZEND_COLD zend_function *zend_closure_get_constructor(zend_object *object
 }
 /* }}} */
 
+static zend_always_inline int zend_closure_compare_fakes(zend_closure *lhs, zend_closure *rhs) { /* {{{ */
+	if (Z_TYPE(lhs->this_ptr) == IS_OBJECT && Z_TYPE(rhs->this_ptr) == IS_OBJECT) {
+		if (Z_OBJ(lhs->this_ptr) != Z_OBJ(rhs->this_ptr)) {
+			return ZEND_UNCOMPARABLE;
+		}
+	}
+
+	if (lhs->called_scope != rhs->called_scope) {
+		return ZEND_UNCOMPARABLE;
+	}
+
+	if (lhs->func.type != rhs->func.type) {
+		return ZEND_UNCOMPARABLE;
+	}
+
+	if (lhs->func.type == ZEND_USER_FUNCTION) {
+		return lhs->func.op_array.opcodes != rhs->func.op_array.opcodes;
+	}
+
+	if (lhs->func.common.scope != rhs->func.common.scope) {
```

Comment:
Might be fine, but I'm not sure why this is checked for internal only? Shouldn't opcodes be identical when it's a trait? And potentially only vary in scope.

Reply:
Good call, scope should always match.